### PR TITLE
Support direct constants in inline assembly.

### LINKED ISSUE #3776: Constant variables in inline assemby
> TypeError: Constant variables not supported by inline assembly.

Is there a reason for this error? You could evaluate the constant and then copy value to assembly in compile time, right?

### PR DESCRIPTION
**Pull request by chriseth**
Fixes #3776

### CODE REVIEW

File: libsolidity/codegen/ContractCompiler.cpp [hunk 1]
```diff
@@ -629,8 +629,46 @@ bool ContractCompiler::visit(InlineAssembly const& _inlineAssembly)
...
+				if (variable->isConstant())
+				{
+					u256 value;
+					if (variable->value()->annotation().type->category() == Type::Category::RationalNumber)
+					{
+						value = dynamic_cast<RationalNumberType const&>(*variable->value()->annotation().type).literalValue(nullptr);
+						if (FixedBytesType const* bytesType = dynamic_cast<FixedBytesType const*>(variable->type()))
+							value = value << (256 - 8 * bytesType->numBytes());
+						else
+							solAssert(variable->type()->category() == Type::Category::Integer, "");
```
Comment (leonardoalt) [inline libsolidity/codegen/ContractCompiler.cpp:641]: What about fixed point?

Comment (chriseth) [reply libsolidity/codegen/ContractCompiler.cpp:641]: Are you saying fixed points should be implemented here or they should be prevented from passing through this code?

Comment (leonardoalt) [reply libsolidity/codegen/ContractCompiler.cpp:641]: My question is rather what happens then. Does it reach an assert(false)?

Comment (chriseth) [reply libsolidity/codegen/ContractCompiler.cpp:641]: yes, it asserts here.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**